src/lockscreen.js

```diff
@@ -42,6 +49,39 @@ var Lockscreen = GObject.registerClass({
         this.notify('locked');
     }
 
+    get key() {
+        return this._key;
+    }
+
+    set key(key) {
+        if ('_key' in this && this._key === key)
+            return;
+        this._key = key;
+        this._updateLockState();
+    }
+
+    _onChanged() {
```

Comment:
Looking at how this signal is used, it looks to me that you'd want a signal with details, with the detail being the key itself. So here the lockscreen could connect to `changed::key` and avoid being waken up for any other change.